File: src/test/testdata/inspections/unsafeIterators/LongMap.java
package main;

import com.badlogic.gdx.utils.LongMap;

class LongMapTest {

  private LongMap<String> map = new LongMap();

  void init() {
    map.put(1L, "One");
    map.put(2L, "Two");
    map.put(3L, "Three");
  }

  void print() {
    for (LongMap.Entry e: <warning>map</warning>) {
      System.out.println(e.value);
    }
  }

  void print2() {
    for (LongMap.Entry<String> e: new LongMap.Entries<String>(map)) {
      System.out.println(e.value);
    }
  }

  void test() {
    long[] i = <warning>map.keys()</warning>.toArray().toArray();
    LongMap map2 = map;
    Object o = map2;
    <warning>((LongMap)o).entries()</warning>;
    <warning>((LongMap)o).iterator()</warning>;
    <warning>((LongMap)o).values()</warning>;
  }

}

File: src/test/testdata/inspections/unsafeIterators/LongMap.kt
package main

import com.badlogic.gdx.utils.LongMap

@Suppress("UNUSED_VARIABLE", "UNCHECKED_CAST", "USELESS_CAST")
internal class LongMapTestKotlin {

  private val map = LongMap<String>()

  fun init() {
    map.put(1L, "One")
    map.put(2L, "Two")
    map.put(3L, "Three")
  }

  fun print() {
    for (e in <warning>map</warning>) {
      println(e.value)
    }
  }

  fun print2() {
    for (e in LongMap.Entries<String>(map)) {
      println(e.value)
    }
  }

  fun test() {
    val i = <warning>map.keys()</warning>.toArray().toArray()
    val map2 = map
    val o: Any = map2
    <warning>(o as LongMap<String>).entries()</warning>
    <warning>(o as LongMap<String>).iterator()</warning>
    <warning>(o as LongMap<String>).values()</warning>

    val map3 = if (true==false) map2 else null

    map3?.let {
      <warning>it.keys()</warning>
      it.let { bla ->
        <warning>bla.values()</warning>
      }
    }
  }

}
